Elfie/Xsv.Test/Sanitize/SanitizerTests.cs

```diff
@@ -0,0 +1,154 @@
+
+            // Verify hashes vary based on hashKey
+            Assert.AreNotEqual(Hashing.Hash(String8.Empty, 1), Hashing.Hash(String8.Empty, 2));
+            Assert.AreNotEqual(Hashing.Hash(one, 1), Hashing.Hash(one, 2));
+            Assert.AreNotEqual(Hashing.Hash(two, 1), Hashing.Hash(two, 2));
+
+            // Verify hashes vary based on value
+            Assert.AreNotEqual(Hashing.Hash(String8.Empty, 3), Hashing.Hash(one, 3));
+            Assert.AreNotEqual(Hashing.Hash(String8.Empty, 3), Hashing.Hash(two, 3));
+            Assert.AreNotEqual(Hashing.Hash(one, 3), Hashing.Hash(two, 3));
+        }
+
+        private static int ExtractCount(uint hash, int countLimit)
+        {
+            int partsExtracted = 0;
+            while (hash > 0)
+            {
+                int part = Hashing.Extract(ref hash, countLimit);
+                partsExtracted++;
+            }
+            return partsExtracted;
+        }
+
+        [TestMethod]
+        public void Sanitizer_Mapper_Basics()
+        {
+            SanitizerProvider p = new SanitizerProvider();
+
+            // Verify each known mapper behaves as expected
+            foreach(string mapperName in p.MapperTypes())
+            {
+                Trace.WriteLine(mapperName);
+                Mapper_Basics(p.Mapper(mapperName));
+            }
+
+            // Verify extensibility added the working mapper and not the two invalid ones
+            Assert.IsTrue(p.MapperTypes().Contains("Test"));
+            Assert.IsFalse(p.MapperTypes().Contains("noEmptyCtor"));
+            Assert.IsFalse(p.MapperTypes().Contains("typeNotFound"));
+
+            // Verify exception when an unknown mapper is requested
+            try
+            {
+                p.Mapper("MapperTypeWhichIsn'tDefined");
+                Assert.Fail("SanitizerProvider.Mapper() should throw when asked for an unknown mapper.");
+            }
+            catch(UsageException)
+            {
+                // Pass
+            }
+        }
+
+        private static void Mapper_Basics(ISanitizeMapper mapper)
+        {
+            // Verify mapper doesn't throw for min and max value and produces different values
+            Assert.AreNotEqual(mapper.Generate(uint.MinValue), mapper.Generate(uint.MaxValue));
+
+            // Verify no collisions for 10k values (using a consistent seed for repeatability)
+            Random r = new Random(0);
+            HashSet<uint> hashes = new HashSet<uint>();
+            HashSet<string> results = new HashSet<string>();
+            for(int i = 0; i < 10000; ++i)
+            {
+                // If the hash isn't a duplicate, verify the result isn't either
+                uint hash = (uint)r.Next();
+                if (hashes.Add(hash))
+                {
+                    Assert.IsTrue(results.Add(mapper.Generate(hash)), $"Mapper {mapper.GetType().Name} produced collision for {i}.");
```

Comment:
Jeff: How about while(hashes.Count < 10000) to make sure you're getting enough cases?